NDTensors/src/Unwrap/test/runtests.jl

```diff
@@ -138,4 +138,33 @@ include("../../../test/device_list.jl")
   x = dev(randn(elt, 4, 4))
   permutedims!(expose(y), expose(x), (2, 1))
   @test NDTensors.cpu(y) == transpose(NDTensors.cpu(x))
+
+  ##########################################
+  ### Testing an issue with CUDA&Metal transpose/adjoint mul
+  A = dev(randn(Float64, (3, 2)))
+  B = dev(randn(Float64, (3, 4)))
+  C = dev(randn(Float64, (4, 2)))
+  Cp = copy(C)
+  if (dev == NDTensors.cu)
+    CUDA.allowscalar(false)
+  end
```

Comment:
This seems like an odd place to put this.

Reply:
In principle all of the tests will not use scalar but they are at the moment. I removed this from the PR

Reply:
I see, thanks.

Would it be difficult to go ahead and set `CUDA.allowscalar(false)` globally for tests (i.e. in `device_list.jl`) and then use `@allowscalar` in places we specifically want to allow it or would that lead to too many things to have to fix? Seems like we should do that ASAP or else we will be dealing with a wack-a-mole of scalar indexing issues (i.e. accidentally introducing them and then having to fix them).

Reply:
I am not sure but since we are dealing with these situations fairly often I can look into it with this PR. I don't believe it will be difficult but I will let you know if I find otherwise!